### PR TITLE
Merge refactor branch into main branch

### PR DESCRIPTION
**Pull request by ptelang**

### CODE REVIEW

**Review by courtneyambrozic — APPROVED** [review]
This looks great! Thanks to all that made changes to clean this up.

File: cvpy/biomedimage/BiomedImageTable.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+        # Load the actionsets
+        self.connection.loadactionset('image')
+        self.connection.loadactionset('biomedimage')
+        self.connection.loadactionset('fedsql')
+
+    def fetch_image_array(self, n: int = 0, qry: str = None, image: str = '_image_', dim: str = '_dimension_',
+                          res: str = '_resolution_', ctype: str = '_channelType_', ccount: int = 1) -> numpy.ndarray:
+        """
+        Fetch image array from an Imagetable.
```
Comment (kybowm) [inline cvpy/biomedimage/BiomedImageTable.py:78]: ImageTable

**Review by kybowm — COMMENTED** [review]
I might not fully understand the implications of this refactoring. Do we intend users to interact with the ImageTable class directly or only by using the NaturalImageTable and BiomedImageTable class?

When I was reading docstrings, I was surprised that many NatualImageTable methods and BiomedImageTable methods returned ImageTables. As a user, I would instinctively expect NaturalImageTable -> NaturalImageTable so you could repeatedly apply methods. 

That made me think that I might misunderstand the design. I don't want to give bad suggestions based on faulty assumptions.

File: cvpy/image/NaturalImageTable.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+import numpy as np
+from typing import List, Dict
+from warnings import warn
+from swat import CASTable
+from cvpy.base.ImageTable import ImageTable
+from cvpy.utils.RandomNameGenerator import RandomNameGenerator
+from enum import *
+
+
+class NaturalImageTable(ImageTable):
```
[the remark this answers is not in the document]
Comment (ptelang) [reply cvpy/image/NaturalImageTable.py:32]: Users are generally expected to use ImageTable.load() or ImageTable.from_table() functions to create instances of NaturalImageTable or BiomedImageTable. The load() and from_table() functions can figure out what type of images are in the table and will create either NaturalImageTable or BiomedImageTable.

I updated the docstrings of ImageTable and NaturalImageTable per your suggestion.

File: cvpy/biomedimage/BiomedImageTable.py (hunk 1 above)
Comment (ptelang) [reply cvpy/biomedimage/BiomedImageTable.py]: I updated the description.

**Comment by ptelang** [comment]
@kybowm, Good points. See my responses inline. 

> I might not fully understand the implications of this refactoring. Do we intend users to interact with the ImageTable class directly or only by using the NaturalImageTable and BiomedImageTable class?
=> Users will typically use ImageTable.load() or ImageTable.from_table() functions to create instances of NaturalImageTable and BiomedImageTable classes. They will then call functions on those instances.

> When I was reading docstrings, I was surprised that many NatualImageTable methods and BiomedImageTable methods returned ImageTables. As a user, I would instinctively expect NaturalImageTable -> NaturalImageTable so you could repeatedly apply methods. That made me think that I might misunderstand the design. I don't want to give bad suggestions based on faulty assumptions.
=> I modified the functions in NaturalImageTable and BiomedImageTable classes to return instances of the NaturalImageTable and BiomedImageTable classes.

**Comment by kybowm** [comment]
@ptelang , awesome! I'm glad I asked. That's a slick design to let ImageTable decide. With your updates, I have no further comments. Thanks!

**Pull request merged**